src/hooks/useEventBusForChat/index.ts

```diff
@@ -580,6 +638,7 @@ export type ChatCapsState = {
 
 export type ChatState = {
   chat: ChatThread;
+  chat_cache: ChatThread[];
```

Comment:
Would an object make more sense? since there can't be any duplicates and looking up the chat is 
```js
if (payload.chat.id in chat_cache) { ... } 
```

or if it needs to be an array you could remove some of the loops with the `find` method on the array

Reply:
Ah yes you are right, I will change it to an object